activerecord/lib/active_record/relation.rb

```diff
@@ -643,6 +644,15 @@ def delete_by(*args)
       where(*args).delete_all
     end
 
+    def async
+      spawn.async!
+    end
+
+    def async! # :nodoc:
+      @async = true
```

Comment:
I went for a boolean attribute, but an `AsyncRelation` subclass could make sense if it makes specializing the various method easier. To be seen once and if we move forward.

Reply:
I think that a subclass would be clearer here so if we do move this forward +1 on that change.